tools/xo_rest.py

```
@@ -0,0 +1,139 @@
+            )
+
+    def request(self, method, path, body=None):
+        url = f"{self.base}{path}"
+        data = json.dumps(body).encode() if body is not None else None
+        req = urllib.request.Request(url, data=data, method=method)
+        req.add_header("Cookie", f"authenticationToken={self._token}")
+        if data is not None:
+            req.add_header("Content-Type", "application/json")
+
+        try:
+            with urllib.request.urlopen(req, timeout=self.timeout, context=self._ctx) as resp:
```

Ruff (0.16.3)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)